.github/scripts/e2e_tests_brownfield_use_case.sh

```diff
@@ -0,0 +1,115 @@
+
+# Check if the bucket exists
+log_and_run echo "Checking if bucket $BUCKET_NAME exists"
+aws --endpoint-url "$S3_ENDPOINT" s3api head-bucket --bucket "$BUCKET_NAME"
+log_and_run echo "Bucket $BUCKET_NAME exists!"
+
+log_and_run echo "Applying Bucket Class to use existing bucket..."
+log_and_run kubectl apply -f cosi-examples/brownfield/bucketclass.yaml
+
+log_and_run echo "Manually creating Bucket object with existing bucket..."
+log_and_run kubectl apply -f cosi-examples/brownfield/bucket.yaml
+
+log_and_run echo "Applying Bucket Claim referencing the Bucket object..."
+log_and_run kubectl apply -f cosi-examples/brownfield/bucketclaim.yaml
+
+log_and_run echo "Applying Bucket Access Class..."
+log_and_run kubectl apply -f cosi-examples/brownfield/bucketaccessclass.yaml
+
+log_and_run echo "Applying Bucket Access..."
+log_and_run kubectl apply -f cosi-examples/brownfield/bucketaccess.yaml
+
+log_and_run echo "Verifying brownfield-bucket-secret in the default namespace..."
+SECRET_JSON="$(kubectl get secret "$SECRET_NAME" --namespace "$NAMESPACE" -o json)"
+
+# Decode the Base64 encoded BucketInfo
+BUCKET_INFO_BASE64="$(echo "$SECRET_JSON" | jq -r '.data.BucketInfo')"
+BUCKET_INFO_JSON="$(echo "$BUCKET_INFO_BASE64" | base64 --decode)"
+
+log_and_run echo "Decoded BucketInfo: $BUCKET_INFO_JSON"
+
+# Extract values to verify
+ACTUAL_BUCKET_NAME=$(echo "$BUCKET_INFO_JSON" | jq -r '.spec.bucketName')
+ACTUAL_ENDPOINT=$(echo "$BUCKET_INFO_JSON" | jq -r '.spec.secretS3.endpoint')
+ACTUAL_REGION=$(echo "$BUCKET_INFO_JSON" | jq -r '.spec.secretS3.region')
+ACTUAL_ACCESS_KEY_ID=$(echo "$BUCKET_INFO_JSON" | jq -r '.spec.secretS3.accessKeyID')
+ACTUAL_ACCESS_SECRET_KEY=$(echo "$BUCKET_INFO_JSON" | jq -r '.spec.secretS3.accessSecretKey')
+ACTUAL_PROTOCOLS=$(echo "$BUCKET_INFO_JSON" | jq -c '.spec.protocols')
+
+# Verify bucketName
+if [[ "$ACTUAL_BUCKET_NAME" != "$BUCKET_NAME" ]]; then
+  log_and_run echo "Bucket name mismatch! Expected: $BUCKET_NAME, Found: $ACTUAL_BUCKET_NAME"
+  exit 1
+fi
+
+# Verify endpoint
+EXPECTED_ENDPOINT="$S3_ENDPOINT"
+if [[ "$ACTUAL_ENDPOINT" != "$EXPECTED_ENDPOINT" ]]; then
+  log_and_run echo "Endpoint mismatch! Expected: $EXPECTED_ENDPOINT, Found: $ACTUAL_ENDPOINT"
+  exit 1
+fi
+
+# Verify region
+if [[ "$ACTUAL_REGION" != "$REGION" ]]; then
+  log_and_run echo "Region mismatch! Expected: $REGION, Found: $ACTUAL_REGION"
+  exit 1
+fi
+
+# Verify accessSecretKey exists
+if [[ -z "$ACTUAL_ACCESS_KEY_ID" ]]; then
+  log_and_run echo "AccessSecretKey is empty!"
+  exit 1
+fi
+
+# Verify accessSecretKey exists
+if [[ -z "$ACTUAL_ACCESS_SECRET_KEY" ]]; then
+  log_and_run echo "AccessSecretKey is empty!"
+  exit 1
+fi
+
+# Verify protocol
+EXPECTED_PROTOCOLS='["s3"]'
+if [[ "$ACTUAL_PROTOCOLS" != "$EXPECTED_PROTOCOLS" ]]; then
+  log_and_run echo "Protocols mismatch! Expected: $EXPECTED_PROTOCOLS, Found: $ACTUAL_PROTOCOLS"
+  exit 1
+fi
+
```

Comment:
Perhaps we could perform some action on the bucket since we have everything we need at this stage, even just a `HeadBucket` or `ListObjects`, **if** that's not too complicated.

Reply:
I added a test in the lastest [review commit](https://github.com/scality/cosi-driver/pull/68/commits/89e0dfc1730311709252312a4dffc4fc35ee733a)